Framework/Kernel/src/NexusHDF5Descriptor.cpp

```diff
@@ -25,7 +24,48 @@ namespace Mantid::Kernel {
 /// PRIVATE
 namespace {
 
-void getGroup(H5::Group groupID, std::map<std::string, std::set<std::string>> &allEntries) {
+/// Size of HDF magic number
+const size_t HDFMagicSize = 4;
```

Comment:
Should this go in NexusDescriptor instead? Also would we be better of using the hdf4/hdf5 APIs for this check. [Example using C from napi](https://github.com/mantidproject/mantid/blob/main/Framework/NexusCpp/src/napi.cpp#L367-L378).